src/plugins/ResizeReorder/ResizerKnob.js

```diff
@@ -0,0 +1,170 @@
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ResizerKnob
+ * @typechecks
+ */
+
+import React from 'react';
+import cx from 'cx';
+import joinClasses from 'joinClasses';
+import FixedDataTableEventHelper from 'FixedDataTableEventHelper';
+import ResizerLine from 'ResizerLine';
+import clamp from 'clamp';
+import DOMMouseMoveTracker from 'DOMMouseMoveTracker';
+
+
+class ResizerKnob extends React.Component {
+
+  initialState = {
+    isColumnResizing: undefined,
+    instance: undefined,
+    mouseXCoordinate: 0,
+    totalDisplacement: 0
+  };
+
+  state = { ...this.initialState };
+
+  constructor(props) {
+    super(props);
+    this.trackMouse();
+  }
+
+
+  componentDidMount() {
+    this.setState({
+      instance: this.curRef
+    });
+  }
+
+  render() {
+    const dragKnob =
+      <ResizerLine
+        height={this.props.resizerLineHeight}
+        visible={!!this.state.isColumnResizing}
+        instance={this.state.instance}
+        xCoordinate={this.state.resizerLineXCoordinate}
+      />;
+
+    return (
+      <div
+        className={cx('fixedDataTableCellLayout/columnResizerContainer')}
+        ref={(element) => this.curRef = element}
+        style={this.props.columnResizerStyle}
+        onMouseDown={this.onMouseDown}
+        onTouchStart={this.props.touchEnabled ? this.onMouseDown : null}
+        onTouchEnd={this.props.touchEnabled ? this.suppressEvent : null}
+        onTouchMove={this.props.touchEnabled ? this.suppressEvent : null}>
+        {dragKnob}
+        <div
+          className={joinClasses(
+            cx('fixedDataTableCellLayout/columnResizerKnob'),
+            cx('public/fixedDataTableCell/columnResizerKnob'),
+          )}
+          style={this.props.columnResizerStyle}
+        />
+      </div>
+    );
+  }
+
+  /**
+   * Registers event listeners for mouse tracking
+   */
+  trackMouse = () => {
+    this._mouseMoveTracker = new DOMMouseMoveTracker(
+      this.onMouseMove,
+      this.onMouseUp,
+      document.body,
+      this.props.touchEnabled
+    );
+  };
+
+  /**
+   * @param {MouseEvent} ev Mouse down event
+   */
+  onMouseDown = (ev) => {
+    const initialMouseCoordinates = FixedDataTableEventHelper.getCoordinatesFromEvent(ev);
+    this._mouseMoveTracker.captureMouseMoves(ev);
+    this.setState({
+      initialMouseCoordinates,
+      isColumnResizing: true,
+      totalDisplacement: 0,
+      resizerLineXCoordinate: initialMouseCoordinates.x
+    });
+  };
+
+  onMouseUp = () => {
+    const { minWidth, maxWidth } = this.getMinMaxWidth(this.props.minWidth, this.props.maxWidth);
+    const newWidth = clamp(this.props.width + this.state.totalDisplacement, minWidth, maxWidth);
+    this.props.onColumnResizeEnd(newWidth, this.props.columnKey);
```

Comment:
I can't think if there'll be any data races here, but it might be safer to first reset column resizing state, and then on the same setState call, specify onColumnResizeEnd as the callback.

```
this.setState({
  isColumnResizing: false,
  totalDisplacement: 0
}, () => {
  this.props.onColumnResizeEnd(newWidth, this.props.columnKey);
});
```

Reply:
Done